src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumDeserializationTest.java

```diff
@@ -206,7 +206,38 @@ private Enum2309(String value) {
         public String toString() {
             return value;
         }
-    }        
+    }
+
+    // for [databind#2352]: Support aliases on enum values
+    enum MyEnum2352_1 {
+        A,
+        @JsonAlias({"singleAlias"})
+        B,
+        @JsonAlias({"multipleAliases1", "multipleAliases2"})
+        C
+    }
+    // for [databind#2352]: Support aliases on enum values
+    enum MyEnum2352_2 {
+        A,
+        @JsonAlias({"singleAlias"})
+        B,
+        @JsonAlias({"multipleAliases1", "multipleAliases2"})
+        C;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase();
+        }
+    }
+    // for [databind#2352]: Support aliases on enum values
+    enum MyEnum2352_3 {
+        A,
+        @JsonEnumDefaultValue
+        @JsonAlias({"singleAlias"})
+        B,
+        @JsonAlias({"multipleAliases1", "multipleAliases2"})
+        C
```

Comment:
Looks good, but I think there's need for at least one more case: Enum value sub-classes... I've been burnt by them a few times.

For sub-classing simply need to use inner-class notation (may define base implementation too), something like

```
enum SubEnum {
     A(1),
     B(2) {
     }

     protected SubEnum(int dummy) { }
```

The reason this is important is that structurally this will differ from simple case and sometimes finding Enum values etc requires bit more complex handling.